crates/goose/src/systems/non_developer.rs

```diff
@@ -0,0 +1,971 @@
+
+        let cache_tool = Tool::new(
+            "cache",
+            indoc! {r#"
+                Manage cached files and data:
+                - list: List all cached files
+                - view: View content of a cached file
+                - delete: Delete a cached file
+                - clear: Clear all cached files
+            "#},
+            json!({
+                "type": "object",
+                "required": ["operation"],
+                "properties": {
+                    "operation": {
+                        "type": "string",
+                        "enum": ["list", "view", "delete", "clear"],
+                        "description": "The operation to perform"
+                    },
+                    "path": {
+                        "type": "string",
+                        "description": "Path to the cached file for view/delete operations"
+                    }
+                }
+            }),
+        );
+
+        // Create cache directory in user's home directory
+        let cache_dir = dirs::cache_dir()
+            .unwrap_or_else(|| PathBuf::from("/tmp"))
+            .join("goose")
+            .join("non_developer");
+        fs::create_dir_all(&cache_dir).unwrap_or_else(|_| {
+            println!(
+                "Warning: Failed to create cache directory at {:?}",
+                cache_dir
+            )
+        });
+
+        let instructions = formatdoc! {r#"
+            You are a helpful assistant to a power user who is not a professional developer, but you may use devleopment tools to help assist them.
+            The user may not know how to break down tasks, so you will need to ensure that you do, and run things in batches as needed.
+            The NonDeveloperSystem helps you with common tasks like web scraping,
+            data processing, and automation without requiring programming expertise,
+            supplementing the Developer System.
+
+            You can use scripting as needed to work with text files of data, such as csvs, json, or text files etc.
+            Using the developer system is allowed for more sophisticated tasks or instructed to (js or py can be helpful for more complex tasks if tools are available).
+            
+            Accessing web sites, even apis, may be common (you can use bash scripting to do this) without troubling them too much (they won't know what limits are).
+
+            Try to do your best to find ways to complete a task without too many quesitons or offering options unless it is really unclear, find a way if you can. 
+            You can also guide them steps if they can help out as you go along.
+
+            Here are some extra tools:
+
+            web_scrape
+              - Fetch content from websites and APIs
+              - Save as text, JSON, or binary files
+              - Content is cached locally for later use
+              - if website doesn't support it find an alternative way.
+
+            data_process
```

Comment:
just a thought: i wonder if we can get away without this tool and just prompt it for the operations and show a few example use cases:
```
pattern matching: rg pattern README.md
sort unique: cat file.txt | sort | uniq
extracting 2nd column in csv: awk -F "," '{ print $2}'
```
reason for doing it like this would be to cover a much broader set of data processing use cases by utilizing the shell

Reply:
rg isn't installed by default however (as non developers) - so would need to be careful (but standard shell is probably fine - and yeah I think could do without it)

Reply:
yeah removed this and added more prompting